src/split-panel/implementation.tsx

```diff
@@ -179,7 +180,7 @@ export function SplitPanelImplementation({
 
   const mergedRef = useMergeRefs(splitPanelRefObject, __internalRootRef);
 
-  if (closeBehavior === 'hide' && !isOpen) {
+  if (closeBehavior === 'hide' && !isOpen && animationDisabled) {
```

Comment:
This is the key change: unless animations are still disabled render the split panel even if it is hidden (at least up to the relevant wrappers which contain the animated size changes), so that we can animate on enter and leave. Then the implementation of the side and bottom split panel respectively take care of not rendering internal content or using ARIA-hidden when hidden.